loader/src/sources/riteaid/api.js

```diff
@@ -40,22 +126,24 @@ async function queryState(state, rateLimit = null) {
 
   if (rateLimit) await rateLimit.ready();
 
-  const body = await httpClient({
+  const response = await httpClient({
     url: RITE_AID_URL,
     headers: { "Proxy-Authorization": "ldap " + RITE_AID_KEY },
     searchParams: { stateCode: state },
-  }).json();
-
-  if (body.Status !== "SUCCESS") {
-    console.error(body.Status);
-    console.error(body.ErrCde);
-    console.error(body.ErrMsg);
-    console.error(body.ErrMsgDtl);
+    responseType: "json",
+  });
 
-    throw new Error("RiteAid API request failed");
+  if (response.body.Status !== "SUCCESS") {
+    throw new RiteAidApiError(response);
   }
 
-  return body.Data.providerDetails.map(formatStore);
+  assertSchema(
+    riteAidWrapperSchema,
+    response.body,
+    "Response did not match schema"
+  );
```

Comment:
I made separate schemas for the API response wrapper and the actual objects because:

1. It’s possible only some locations could get out of spec, so this lets us avoid breaking everything in that case. (OTOH, maybe we *should* break everything in that case? Also, not sure this scenario is at all likely to occur.)

2. From a practical standpoint, the error message lists every error, and if there is a systemic change, the error message would be incredibly long because it repeats the same things hundreds or thousands of times (one for every location object) if there was just one schema instead of a separate one for the locations that we check when formatting on each location.

    Of course, that could be mitigated by limiting how many errors we include in the actual error message here: https://github.com/usdigitalresponse/univaf/blob/19f0f923aa840bf91cfa959db6dc56a296d131e7/loader/src/schema-validation.js#L10-L15